### PR TITLE
WIP DO NOT MERGE StepFunctions Dynamic Parallelism

### PR DESCRIPTION
**Pull request by rhboyd**
would like to get some early feedback
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by RomainMuller** [comment]
@rhboyd - you might be interested to learn that [GitHub now supports submitting "WIP" PR's as a first-class concept](github wip pr), so they *cannot* be merged before you "convert out to regular PR". 😬

**Comment by rhboyd** [comment]
That’s great to hear. I didn’t now that.

**Comment by rix0rrr** [comment]
Duplicate: https://github.com/aws/aws-cdk/pull/4145

Will need some time to go over both and compare.

File: packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+     */
+    public next(next: IChainable): Chain {
+        super.makeNext(next.startState);
+        return Chain.sequence(this, next);
+    }
+
+    /**
+     * Define one or more graphs to run in map
+     */
+    public dynamicBranch(graph: IChainable): Map {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts]: Can we call it `iterator` like in the official service docs?

File: packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ * The Result of a Map state is an array of the results of its substatemachines.
+ */
+export class Map extends State implements INextable {
+    public readonly endStates: INextable[];
+
+    /**
+     * Usually, State Properties are contained in the state.ts file, but maxConcurrency
+     * only exists in this one state (for now).
+     */
+    protected readonly maxConcurrency: string;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts]: This does not get set, does it?

**Comment by rix0rrr** [comment]
Closing in favor of https://github.com/aws/aws-cdk/pull/4145

Thanks for the enthusiasm!

**Pull request closed without merging**